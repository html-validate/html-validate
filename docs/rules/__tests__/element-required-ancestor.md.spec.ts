import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<area>`;
markup["correct"] = `<map>
    <area>
</map>`;

describe("docs/rules/element-required-ancestor.md", () => {
	it("inline validation: incorrect", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"element-required-ancestor":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"element-required-ancestor":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
});
