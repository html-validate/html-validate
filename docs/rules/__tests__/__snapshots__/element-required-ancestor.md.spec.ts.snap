// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`docs/rules/element-required-ancestor.md inline validation: correct 1`] = `[]`;

exports[`docs/rules/element-required-ancestor.md inline validation: incorrect 1`] = `
[
  {
    "errorCount": 1,
    "filePath": "inline",
    "messages": [
      {
        "column": 2,
        "context": {
          "ancestor": [
            "<map>",
            "<template>",
          ],
          "child": "<area>",
        },
        "line": 1,
        "message": "<area> element requires a <map> or <template> ancestor",
        "offset": 1,
        "ruleId": "element-required-ancestor",
        "ruleUrl": "https://html-validate.org/rules/element-required-ancestor.html",
        "selector": "area",
        "severity": 2,
        "size": 4,
      },
    ],
    "source": "<area>",
    "warningCount": 0,
  },
]
`;
